### PR TITLE
Initial commit for the Enterprise SONiC "prefix_list" resource module:

### PR DESCRIPTION
**Pull request by kerry-meyer**
##### SUMMARY

- Provide data definitions (argspec), new code modules, and additions/changes
  to existing files in the Enterprise SONiC Ansible collection to implement
  "facts" collection and configuration for the "merged" and "deleted" states
  applied to the "sonic_prefix_list" resource module.

- Provide "module" examples for adding, modifying, and deleting prefix list
  configuration.

- Define regression test cases to verify correct functionality of the
  prefix list resource module.

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

sonic_prefix_list
### Test Results

- Regression test HTML report:
[prefix_list_PR_regression-2022-07-18-13-13-08.pdf](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/9136057/prefix_list_PR_regression-2022-07-18-13-13-08.pdf)

- Regression test detailed log:
[prefix_list_PR_detailed_regression.log](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/9136059/prefix_list_PR_detailed_regression.log)

- Display of "show running-configuration" after merging test prefix_list playbook commands using the proposed change set:
[prefix_list_PR_show_running_config.txt](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/9136111/prefix_list_PR_show_running_config.txt)

```paste below

```

### CODE REVIEW

File: tests/regression/roles/sonic_prefix_lists/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+- name: "Test {{ module_name }} started ..."
+  include_tasks: tasks_template.yaml
+  loop: "{{ tests }}"
+
+- name: "Cleanup {{ module_name }} started ..."
+  include_tasks: cleanup_tests.yaml
+
+# - name: "Test negative {{ module_name }} started ..."
+#   include_tasks: negative_tasks_template.yaml
+#   loop: "{{ negative_tests }}"
```
Comment (stalabi1) [inline tests/regression/roles/sonic_prefix_lists/tasks/main.yml:15]: Should these commented out lines be removed?

File: tests/regression/roles/sonic_prefix_lists/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+#                 safi: unicast
+#                 allowas_in:
+#                   value: 11
+#                 route_map:
+#                   - name: rmap_reg1
+#                     direction: in
+#                   - name: rmap_reg1
+#                     direction: out
+#                 route_reflector_client: true
+#                 route_server_client: true
```
Comment (stalabi1) [inline tests/regression/roles/sonic_prefix_lists/defaults/main.yml:36]: Did you intend to leave these commented lines?

Comment (kerry-meyer) [reply tests/regression/roles/sonic_prefix_lists/defaults/main.yml]: No. (I left them for myself to check at the end as a possible action item reminder, but they are no longer needed.)

I have deleted them along with some other unused lines. Thanks for catching this.

File: tests/regression/roles/sonic_prefix_lists/tasks/main.yml (hunk 1 above)
Comment (kerry-meyer) [reply tests/regression/roles/sonic_prefix_lists/tasks/main.yml]: Yes. I have deleted them.

File: plugins/modules/sonic_prefix_lists.py [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+options:
+  config:
+    description:
+      - Specifies a list of prefix set configuration dictionaries
+    type: list
+    elements: dict
+    suboptions:
+      name:
+        description:
+          - name of a prefix set (a list of prefix entries)
```
Comment (stalabi1) [inline plugins/modules/sonic_prefix_lists.py:49]: Please capitalize the description for consistency.

Comment (kerry-meyer) [reply plugins/modules/sonic_prefix_lists.py]: Done.

**Review by stalabi1 — APPROVED** [review]
I have reviewed and approve of these changes.

**Pull request merged**